src/qml/Legend.qml

```diff
@@ -148,7 +148,7 @@ ListView {
       onClicked: {
           var item = table.itemAt(table.contentX + mouse.x, table.contentY + mouse.y)
           if (item) {
-              if (item.vectorLayer) {
+              if (item.vectorLayer && item.vectorLayer.isValid()) {
```

Comment:
@m-kuhn is there any indication to the user that the layer is invalid and might experience problems with it?

Reply:
there is now a bad layer dialog on startup.
an indicator icon would be even nicer.

Reply:
+1 for indicator icon, or at least a message in the layer properties

Reply:
how does it look to you in gray?

Reply:
@m-kuhn , gray is also used when hidden layers. Could we use something else? We could a semi opaque Theme red color.

Reply:
@nirvn is right about gray, but I guess reddish will grab the attention way too much. If colors don't work, why not icon and explanation in layer properties, similar to the locked layers? If the layer is broken, there is no need to show any other icon, so visually it would not be too heavy.

Reply:
Well, if a layer is broken, it's not bad to draw attention to it :)

Reply:
Well, there is the project welcome screen listing all the broken layers, the user already knows something went wrong and they continue using the project, despite the broken layers. 
But having it red might behave themselves and better fix layers for the next time so I am now convinced dull red is the color :)

Reply:
not sure, red screams for attention. And there is possibly nothing you can do to get rid of this out in the fields. I'll experiment with an icon.

Reply:
See the update below. Icon and text added.

Reply:
Not sure if it must be grayed out (not to be confused with hidden layers). Other than that, looks great.

Reply:
It is hidden, there is no data around to be worked with. I think hiding it away is quite intuitive for "don't work with this".